datasets.py

```diff
@@ -39,6 +40,8 @@ def read_transcript(data_path):
 		self.examples = [list(g) for data_path in data_paths for k, g in itertools.groupby(sorted(read_transcript(data_path), key = transcripts.sort_key), key = transcripts.group_key)]
 		self.examples = list(sorted(self.examples, key = duration))
 		self.examples = list(filter(lambda example: (min_duration is None or min_duration <= duration(example)) and (max_duration is None or duration(example) <= max_duration), self.examples))
+		self.examples = [e for e in self.examples if os.path.basename(e[0]['audio_path']) not in set(exclude)]
+		random.shuffle(self.examples)
```

Comment:
You mentioned `random.shuffle(self.examples) for randomize processing of youtube files at transcribe step.`

Why is this randomization needed?

Even if it's needed, this shouldn't be handled by changing the Dataset constructor. You can use `shuffle=True` at the DataLoader level or use an appropriate sampler.

Reply:
I switched it because skipped records accumulates. After restarting, we will spend a lot of time to load and skip old records. Since they are at the top of the list.

Reply:
Let's talk about this on tg